src/proposer.ts

```diff
@@ -132,27 +134,14 @@ let isInitialized = false
 
 // ~7 days at ~12s blocks
 const APPROX_7D_BLOCKS = 50400
-/**
- * Validates that a vault ID exists on-chain by checking it is within range
- * [1, nextVaultId - 1]. Throws if invalid or out of range.
- */
-export async function validateVaultIdOnChain(vaultId: number): Promise<void> {
-	// Basic sanity
-	if (!Number.isInteger(vaultId) || vaultId < 1) {
-		throw new Error('Invalid vault ID')
-	}
-	// Ensure contracts are initialized
+// Wrapper to use validator's on-chain vault ID validation with contract dependency
+const validateVaultIdOnChain = async (vaultId: number): Promise<void> => {
```

Comment:
I think we could fully extract these validator functions to `utils/validator.js`, instead of using base validator functions, but I won't be too fussy!